modules/general/acs-email-with-config-publish/main.bicep

```diff
@@ -0,0 +1,61 @@
+@description('The name of the Azure Communication Service resource.')
+param communicationServiceName string
+
+@description('The name of the Email Communication Service resource.')
+param emailServiceName string
+
+@description('The location where the communication and email service stores its data at rest.')
+@allowed([
+  'Africa'
+  'Asia Pacific'
+  'Australia'
+  'Brazil'
+  'Canada'
+  'Europe'
+  'France'
+  'Germany'
+  'India'
+  'Japan'
+  'Korea'
+  'Norway'
+  'Switzerland'
+  'United Arab Emirates'
+  'UK'
+  'United States'
+])
+param dataLocation string
+
+@description('The username for the sender email address. Defaults to "DoNotReply".')
+param senderUsername string = 'DoNotReply'
+
+@description('The name of the key vault where the configuration will be published')
+param keyVaultName string
+
+module acs_email '../acs-email/main.bicep' = {
+  name: 'AcsEmailDeploy'
+  params: {
+    communicationServiceName: communicationServiceName
+    dataLocation: dataLocation
+    emailServiceName: emailServiceName
+    senderUsername: senderUsername
+  }
+}
+
+module set_secrets './set-secrets.bicep' = {
+  name: 'acsEmailSetSecretsDeploy'
+  params: {
+    keyVaultName: keyVaultName
+    communicationServiceName: communicationServiceName
+    domain: acs_email.outputs.domain
+    sendFromEmailAddress: acs_email.outputs.sendFromEmailAddress
+  }
+  dependsOn: [
```

Comment:
Is the explicit `dependsOn` necessary if you're referencing the symbolic name  - `acs_email` - in the argument to `domain`?

Reply:
No it's not. I previously didn't have any reference to `acs_email` and I forgot to remove the `dependsOn`. Good spot